test/solvers/advanced/proximal_utils_test.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for the utility functions for proximal operators."""
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import numpy as np
+import pytest
+
+# Internal
+import odl
+import odl.solvers as odls
+
+from odl.util.testutils import all_almost_equal
+
+# Places for the accepted error when comparing results
+PLACES = 8
+
+
+def test_proximal_translation():
+    """Test for the proximal of a translation: prox[f(. - g)]"""
+
+    # Image space
+    space = odl.uniform_discr(0, 1, 10)
+
+    # Element in the image space where the proximal operator is evaluated
+    translation = space.element(np.arange(10, 20))
+
+    # Parameters
+    lam = 1 / 3
+    step_size = 2.4
+
+    # Factory function returning the proximal operators
+    prox_factory = odls.proximal_l2_squared(space, lam=lam, g=None)
+
+    # Initialize proximal operators
+    prox = odls.proximal_translation(prox_factory, translation)(step_size)
+
+    # Create an element in the space, in which to evaluate the proximals
+    x = space.element(np.arange(-5, 5))
+
+    # Explicit computation:
+    # (x + 2 * step_size * lam * translation) / (1 + 2 * ste_size * lam)
+    x_verify = ((x + 2 * step_size * lam * translation) /
+                (1 + 2 * step_size * lam))
+
+    print(x_verify)
+    print(prox(x))
+
+    assert all_almost_equal(prox(x), x_verify, places=PLACES)
+
+
+def test_proximal_arg_scaling():
+    """Test for the proximal of scaling: prox[f(. * 1/a)]"""
+
+    # Image space
+    space = odl.uniform_discr(0, 1, 10)
+
+    # Factory function returning the proximal operators
+    prox_factory = odls.proximal_l2_squared(space, lam=1)
+
+    step_size = 2.0
+    scaling_param = 3.4
+
+    # Initialize proximal operators
+    prox_1 = prox_factory(step_size * scaling_param**2)
+    prox_2 = odls.proximal_arg_scaling(prox_factory, scaling_param)(step_size)
+
+    # Create an element in the space, in which to evaluate the proximals
+    x = space.element(np.arange(-5, 5))
+
+    # Uses the fact that 1/2||x 1/a||^2 = 1/2a^2 ||x||^2
```

Comment:
Needs to be more complicated, this works for any `a` so doesn't really test that this works.